R/otp-setup.R

```diff
@@ -0,0 +1,290 @@
+#' Build an OTP Graph
+#'
+#' OTP is run in Java and requires Java commands to be typed into the command line.
+#' The function allows the parameters to be defined in R and automatically passed to Java.
+#' This function builds a OTP graph from the Open Street Map and other files.
+#'
+#' @param otp A character string, path to the OTP .jar file
+#' @param dir A character string, path to a directory containing the necessary files, see details
+#' @param memory A positive integer. Amount of memory to assign to the OTP in GB, default is 2
+#' @param router A character string for the name of the router, must match with contents of dir, default "current"
+#' @param analyst Logial, should analyst feature be built, default FALSE
+#' @return
+#' Returns and log messages produced by OTP, and will return the message "Graph built" if sucessfull
+#' @details
+#' The OTP .jar file can be downloaded from https://repo1.maven.org/maven2/org/opentripplanner/otp/
+#'
+#' To build an OTP graph requires the following files to be in the directory
+#' specified by the dir variable.
+#'
+#' /graphs - A sub-directory
+#'   /current - A sub-directory with the name of the OTP router used in 'router' variaible
+#'     osm.pbf - Required, pbf file containing the Open Street Map
+#'     router-config.json - Required, json file containing configations settings for the OTP
+#'     gtfs.zip - Optional, and number of GTFS files with transit timetables
+#'     terrain.tif - Optional, GeoTiff image of terrain map
+#'
+#' The function will accept any file name for the .jar file, but it must be the only .jar file in that directory
+#' OTP can support multiple routers (e.g. different regions), each router must have its own sub-directory in the graphs directory
+#' @examples
+#' otp_build_graph("C:/otp")
+#' @export
+otp_build_graph <- function(otp = NULL,
+                            dir = NULL,
+                            memory = 2,
+                            router = "current",
```

Comment:
@mem48

r.e. `otp_build_graph()`

For the help example, as a minimum need to provide:

`otp_build_graph(otp = "C:/otp", dir = "C:/otp")`

If the JAR file is also at the dir location, would it be nice to only have to specify that, and not otp too?

There's an issue with reaching max.print. As the build process tends to produce many  warnings. So I never got to see the message that graph build was ok. Would it be better to pipe the OTP build output to a log file instead?

Reply:
I went back and forth on having a separate otp and dir location. On balance I think a separation of code and data is preferable. If a user really wants them in the same place then having to complete two variables is a small hurdle. The change I have made is to make `otp` a path to the file not the folder, this avoids the problem of a folder containing more than one jar file and makes the difference between the two variables clearer.

I also change the help example to show that the results of the log can be stored

Reply:
@mem48 OK, that will make the difference between the two clearer.